crates/matrix-sdk-ui/src/timeline/inner/state.rs

```diff
@@ -427,7 +428,14 @@ impl DerefMut for TimelineInnerState {
 
 pub(in crate::timeline) struct TimelineInnerStateTransaction<'a> {
     pub items: ManuallyDrop<ObservableVectorTransaction<'a, Arc<TimelineItem>>>,
-    pub meta: &'a mut TimelineInnerMetadata,
+
+    /// A clone of the previous meta, that we're operating on during the
+    /// transaction, and that will be committed to the previous meta location in
+    /// [`Self::commit`].
+    pub meta: ManuallyDrop<TimelineInnerMetadata>,
```

Comment:
Why is a `ManuallyDrop` needed here? Could you perhaps add a comment explaining this?

Reply:
It's required because we're using a `ManuallyDrop` for the `ObservableVectorTransaction` (I need to take ownership of the `TimelineInnerMetadata` in `commit()`, and I couldn't take full ownership because this struct has a `Drop` impl), but I'm not even sure why we need the `ManuallyDrop` for the transaction here...

Reply:
@jplatte do you recall why we needed the `ManuallyDrop` for the `ObservableVectorTransaction`? If I remove it and pass all the things by ownership (and remove `TimelineInnerStateTransaction::drop`), then all the tests pass...

Reply:
I suspect the `Drop` implementation was there to log a message that the transaction was cancelled; I think this is pretty rare and that eyeball-im would probably log it out anyways, so let's get rid of it.